Av 280 Интеграция авторизации на странице логина

bySLOVE opened the pull request.

Mikhailuu committed:
Added variables

Commit:
Добавил форму логина и стили

Commit:
внес правки

Commit:
Merge branch 'variables' into AV-279

Commit:
layout

Commit:
Merge branch 'AV-279' into AV-280

Commit:
Интеграция авторизации на странице логина

VladislavNH commented:
Можно вынести типы в отдельный файл, в компонентах объединить все в один файл

Mikhailuu commented on src/app/store.ts:
```diff
@@ -0,0 +1,10 @@
+  },
+  middleware: (getDefaultMiddleware) =>
+    getDefaultMiddleware().concat(authApi.middleware),
+});
```
Отсутствуют типы для стора

Commit:
add alias

Commit:
fix

Commit:
Merge branch 'AV-279' into AV-280

Commit:
- Добавлен базовый запрос с логикой обновления access_token при 401
- Сохранение и обновление refresh_token в localStorage
- Выделены типы для запросов и токенов

alyonastartseva commented on src/features/auth/components/LoginForm.tsx:
```diff
@@ -0,0 +1,76 @@
+    setErrorMessage(null);
+    try {
+      const res = await loginUser({ username: values.email, password: values.password }).unwrap();
+      localStorage.setItem('token', res.access_token);
```
давай работу с localStorage вынесем в кастомный хук

Commit:
Создан кастомный хук useAuthToken

bySLOVE committed:
Merge branch 'develop' into AV-280
